### ISSUE TITLE
python run_all.py failed 

### ISSUE THREAD

**Issue by eh2006re** [Unidata/netcdf4-python#294]
Hello,

I am newbie to netCDF4-python. I install HDF5, netCDF4, Cython, numpy and netCDF4-python without 
problem.   "make check" for netCDF4 passed OK
I am using Debian Squeeze for historical reason.   python2.7.5  has been installed from source /usr/local/bin/python2.7.

However,  when I did
.../netcdf4-python-master/test$ python2.7 run_all.py
it failed, see below. 

any suggestion?

thanks for your time

xg
 

===
Script started on Fri 03 Oct 2014 11:16:51 BST
]0;ucam@atcghost: ~/Install-file/downloadSoftware/netcdf4-python-master/testucam@atcghost:~/Install-file/downloadSoftware/netcdf4-python-master/test$ python2.7 run_all.py
not running tst_unicode3.py ...
....E...........FF.E.....................................E......
======================================================================
ERROR: test_select_nc (tst_netcdftime.TestDate2index)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ucam/Install-file/downloadSoftware/netcdf4-python-master/test/tst_netcdftime.py", line 384, in test_select_nc
    millisecs = int(date2num(d,unix_epoch,calendar='proleptic_gregorian'))
  File "utils.pyx", line 124, in netCDF4.date2num (netCDF4.c:4098)
  File "utils.pyx", line 13, in netCDF4._dateparse (netCDF4.c:2740)
ImportError: dateutil module required for accuracy < 1 second

======================================================================
ERROR: runTest (tst_dap.DapTestCase)
testing access of data over http using opendap
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ucam/Install-file/downloadSoftware/netcdf4-python-master/test/tst_dap.py", line 22, in runTest
    ncfile = netCDF4.Dataset(URL)
  File "netCDF4.pyx", line 1474, in netCDF4.Dataset.__init__ (netCDF4.c:21746)
RuntimeError: NetCDF: Unknown file format

======================================================================
ERROR: runTest (tst_types.PrimitiveTypesTestCase)
testing primitive data type
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ucam/Install-file/downloadSoftware/netcdf4-python-master/test/tst_types.py", line 85, in runTest
    assert(str(issue273_data) == str(v2[:]))
  File "netCDF4.pyx", line 3051, in netCDF4.Variable.__getitem__ (netCDF4.c:41971)
  File "netCDF4.pyx", line 3098, in netCDF4.Variable._toma (netCDF4.c:42953)
NotImplementedError: Not implemented for this type

======================================================================
FAIL: runTest (tst_vlen.TestObjectArrayIndexing)
testing vlen variables
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ucam/Install-file/downloadSoftware/netcdf4-python-master/test/tst_vlen.py", line 104, in runTest
    assert fancy_indexed[0] == 'abc'
AssertionError

======================================================================
FAIL: runTest (tst_vlen.VariablesTestCase)
testing vlen variables
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ucam/Install-file/downloadSoftware/netcdf4-python-master/test/tst_vlen.py", line 68, in runTest
    assert_array_equal(datas, vs_alt[:])
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 707, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 636, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 100.0%)
 x: array([[ab, abc, abcd, abcde, abcdef],
       [abcdefg, abcdefgh, abcdefghi, abcdefghij, abcdefghijk],
       [abcdefghijkl, abcdefghijklm, abcdefghijklmn, abcdefghijklmno,...
 y: array([['a', 'a', 'a', 'a', 'a'],
       ['a', 'a', 'a', 'a', 'a'],
       ['a', 'a', 'a', 'a', 'a'],...

----------------------------------------------------------------------
Ran 64 tests in 9.547s

FAILED (failures=2, errors=3)

netcdf4-python version: 1.1.2
HDF5 lib version:       1.8.13
netcdf lib version:     4.3.2
0;ucam@atcghost: ~/Install-file/downloadSoftware/netcdf4-python-master/testucam@atcghost:~/Install-file/downloadSoftware/netcdf4-python-master/test$ exit
exit

Script done on Fri 03 Oct 2014 11:18:10 BST

**Comment by matthew-brett** [comment]
I have just been experimenting with building standalone OSX wheels for netcdf4-python - but I have the same failures.  The full build log (on travis) is here:

https://api.travis-ci.org/jobs/60479196/log.txt?deansi=true

The test errors are:

```
netcdf4-python version: 1.1.7
HDF5 lib version:       1.8.14
netcdf lib version:     4.3.3.1
numpy version           1.6.2
........F......................E.................E...............F.F
======================================================================
ERROR: runTest (tst_netcdftime.netcdftimeTestCase)
testing netcdftime
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_netcdftime.py", line 55, in runTest
    d2 = self.cdftime_mixed.num2date(t1)
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/netcdftime/netcdftime.py", line 819, in num2date
    date = DateFromJulianDay(jd, self.calendar)
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/netcdftime/netcdftime.py", line 310, in DateFromJulianDay
    dayofyr[inc_idx] = dayofyr[inc_idx] + leap[inc_idx]
TypeError: array cannot be safely cast to required type

======================================================================
ERROR: runTest (tst_types.PrimitiveTypesTestCase)
testing primitive data type
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_types.py", line 85, in runTest
    assert(str(issue273_data) == str(v2[:]))
  File "netCDF4.pyx", line 3148, in netCDF4.Variable.__getitem__ (netCDF4.c:44276)
  File "netCDF4.pyx", line 3203, in netCDF4.Variable._toma (netCDF4.c:45464)
NotImplementedError: Not implemented for this type

======================================================================
FAIL: runTest (tst_endian.EndianTestCase)
testing endian conversion capability
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_endian.py", line 103, in runTest
    issue310(self.file)
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_endian.py", line 65, in issue310
    var_big_endian[:].filled())
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/numpy/testing/utils.py", line 800, in assert_array_almost_equal
    header=('Arrays are not almost equal to %d decimals' % decimal))
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/numpy/testing/utils.py", line 636, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 6 decimals

(mismatch 90.0%)
 x: array([   3.14159265,  999.        ,  999.        ,  999.        ,
        999.        ,  999.        ,  999.        ,  999.        ,
        999.        ,  999.        ])
 y: array([  3.14159265e+000,   1.83134720e-317,   1.83134720e-317,
         1.83134720e-317,   1.83134720e-317,   1.83134720e-317,
         1.83134720e-317,   1.83134720e-317,   1.83134720e-317,
         1.83134720e-317])

======================================================================
FAIL: runTest (tst_vlen.TestObjectArrayIndexing)
testing vlen variables
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_vlen.py", line 117, in runTest
    assert fancy_indexed[0] == 'abc'
AssertionError

======================================================================
FAIL: runTest (tst_vlen.VariablesTestCase)
testing vlen variables
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_vlen.py", line 68, in runTest
    assert_array_equal(datas, vs_alt[:])
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/numpy/testing/utils.py", line 707, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python2.7/site-packages/numpy/testing/utils.py", line 636, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 100.0%)
 x: array([[ab, abc, abcd, abcde, abcdef],
       [abcdefg, abcdefgh, abcdefghi, abcdefghij, abcdefghijk],
       [abcdefghijkl, abcdefghijklm, abcdefghijklmn, abcdefghijklmno,...
 y: array([['a', 'a', 'a', 'a', 'a'],
       ['a', 'a', 'a', 'a', 'a'],
       ['a', 'a', 'a', 'a', 'a'],...

----------------------------------------------------------------------
Ran 68 tests in 10.629s

FAILED (failures=3, errors=2)
```

Can I do anything to help debug?   I have several OSX machines you would be welcome to log into.

**Comment by matthew-brett** [comment]
Same failures building current master:

https://s3.amazonaws.com/archive.travis-ci.org/jobs/60486852/log.txt

**Comment by shoyer** [comment]
@matthew-brett that's a pretty old version of numpy -- have you tried updating it?

**Comment by matthew-brett** [comment]
It is an old version - what is the oldest numpy that netcdf should be compatible with?  I noticed an 'install_requires' for numpy 1.3 in setup.py.

Tests currently running with latest numpy: https://travis-ci.org/MacPython/netcdf4-python-wheels/builds/60489772

**Comment by matthew-brett** [comment]
All passing with numpy 1.9.2 : https://travis-ci.org/MacPython/netcdf4-python-wheels/builds/60491417 - trying 1.9.0

**Comment by matthew-brett** [comment]
All good with numpy 1.8.2: https://travis-ci.org/MacPython/netcdf4-python-wheels

Looking back at the tests on 1.7.1 - you actually only have one failure:

```
======================================================================

ERROR: runTest (tst_netcdftime.netcdftimeTestCase)

testing netcdftime

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/Users/travis/build/MacPython/netcdf4-python-wheels/netcdf4-python/test/tst_netcdftime.py", line 55, in runTest

    d2 = self.cdftime_mixed.num2date(t1)

  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python3.4/site-packages/netcdftime/netcdftime.py", line 819, in num2date

    date = DateFromJulianDay(jd, self.calendar)

  File "/Users/travis/build/MacPython/netcdf4-python-wheels/venv/lib/python3.4/site-packages/netcdftime/netcdftime.py", line 310, in DateFromJulianDay

    dayofyr[inc_idx] = dayofyr[inc_idx] + leap[inc_idx]

TypeError: Cannot cast array data from dtype('float64') to dtype('int32') according to the rule 'safe'
```

https://s3.amazonaws.com/archive.travis-ci.org/jobs/60477427/log.txt

**Comment by matthew-brett** [comment]
Maybe the 1.7.1 test error would be easy to fix?  1.7.1 is becoming a fairly common numpy minimum dependency (maybe 1.6 is still more common).

**Comment by jswhit** [comment]
with pull request #405, all tests pass with numpy 1.7.0.
